BDArmory/Control/BDModuleOrbitalAI.cs

```diff
@@ -534,24 +480,24 @@ private IEnumerator PilotLogic()
                         else
                             fc.attitude = toTarget.normalized;
 
-                        complete = FromTo(vessel, targetVessel).magnitude < maxRange || NearIntercept(relVel, minRange);
+                        complete = FromTo(vessel, targetVessel).sqrMagnitude < maxRange * maxRange || NearIntercept(relVel, minRange);
 
-                        yield return new WaitForFixedUpdate();
+                        yield return wait;
                     }
                 }
                 else
                 {
-                    if (hasPropulsion && (relVel.magnitude > firingSpeed || nearInt))
+                    if (hasPropulsion && (relVel.sqrMagnitude > firingSpeed * firingSpeed || nearInt))
                     {
                         currentStatus = "Maneuvering (Kill Velocity)";
                         while (UnderTimeLimit() && targetVessel != null && !complete)
                         {
                             relVel = targetVessel.GetObtVelocity() - vessel.GetObtVelocity();
                             fc.attitude = (relVel + targetVessel.acceleration).normalized;
-                            complete = relVel.magnitude < firingSpeed / 3;
+                            complete = relVel.sqrMagnitude < firingSpeed * firingSpeed / 3;
```

Comment:
This should be `/ 9` to be consistent with the non-squared version (unless you intended to change it).